File: factory-method/c#/ChickenSource.cs
namespace FactoryMethod
{
    public class ChickenSource : AnimalSource
    {
        public override IAnimal FactoryMethod()
        {
            //Visual studio will force you to override FactoryMethod like this
            // because it was marked abstract in the base class (AnimalSource)
            return new Chicken()
            {
                Weight = 500
            };
        }
    }
}

File: factory-method/typescript/src/Factories/ChickenSource.ts
import {IAnimal} from "Interfaces/IAnimal";
import {AnimalSource} from "Factories/AnimalSource"
import {Chicken} from "Animals/Chicken"
export class ChickenSource extends AnimalSource{
  public FactoryMethod() : IAnimal{
    let chicken : Chicken = new Chicken();
    chicken.Weight = 500;
    return chicken;
  }
}